x-pack/plugins/canvas/server/lib/build_es_request.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { buildBoolArray } from './build_bool_array';
+import {
+  CanvasQueryFilter,
+  ElasticsarchTermFilter,
+  ElasticsearchLuceneQueryStringFilter,
+  ElasticsearchTimeFilter,
+} from './filters';
+
+type ElasticsearchFilter =
+  | ElasticsarchTermFilter
+  | ElasticsearchLuceneQueryStringFilter
+  | ElasticsearchTimeFilter;
+
+interface ElementFilter {
+  size: number;
+  and: CanvasQueryFilter[];
+}
+
+export interface EsRequest {
```

Comment:
nit: sometimes you spell out `Elasticsearch`, sometimes you use `Es` and others `ES`.  Personally, I'd go with `ES`, but go with what you prefer and use it everywhere.